File: apps/cli/src/commands/fetch.ts
import { Command } from "commander";
import { getSession } from "../auth/keychain.ts";
import { buildClient } from "../api/client.ts";

export const fetchCommand = new Command("fetch")
  .description("Fetch a URL with stored session cookies and print the response body")
  .argument("<url>", "URL to fetch")
  .action(async (url: string) => {
    const session = getSession();
    if (!session) {
      console.error("Not logged in. Run `uoplan login` first.");
      process.exit(1);
    }

    const { client } = await buildClient(session);
    const res = await client.get(url, { responseType: "text" });
    process.stdout.write(res.body as string);
  });

File: apps/cli-rs/src/commands/fetch.rs
use anyhow::{anyhow, Result};

use crate::api::PeopleSoftClient;
use crate::auth::get_session;
use crate::error::NoCookiesError;

pub async fn run(url: &str) -> Result<()> {
    let session = get_session().ok_or_else(|| anyhow!(NoCookiesError))?;
    let client = PeopleSoftClient::new(session)?;
    let body = client.get(url).await?;
    println!("{}", body);
    Ok(())
}
